### PR TITLE
Feature/te 8674/te 8677 add new endpoints to the application

### PR DESCRIPTION
**Pull request by afahmyspryker**
## PR Description

TBD

### CODE REVIEW

File: saltstack/base/php/extensions.sls [hunk 1]
```diff
@@ -22,8 +22,8 @@ xdebug:
     - require:
       - pkg: xdebug
 
-{{ php_module('xdebug', salt['pillar.get']('php:enable_xdebug', False), 'fpm') }}
-{{ php_module('xdebug', salt['pillar.get']('php:enable_xdebug', False), 'cli') }}
+{{ php_module('xdebug', salt['pillar.get']('php:enable_xdebug', True), 'fpm') }}
```
Comment (stereomon) [inline saltstack/base/php/extensions.sls]: We should probably keep FALSE

File: saltstack/base/php/files/etc/php/7.1/mods-available/xdebug.ini [hunk 2]
```diff
@@ -1,7 +1 @@
-zend_extension=xdebug.so
```
Comment (stereomon) [inline saltstack/base/php/files/etc/php/7.1/mods-available/xdebug.ini:1]: This line needs to stay

File: saltstack/base/php/files/etc/php/7.1/mods-available/xdebug.ini [hunk 3]
```diff
@@ -1,7 +1 @@
...
-xdebug.remote_enable=0
-xdebug.remote_host=10.10.0.1
```
Comment (stereomon) [inline saltstack/base/php/files/etc/php/7.1/mods-available/xdebug.ini:3]: This line needs to stay but must be changed to `xdebug.client_host=10.10.0.1`

File: saltstack/base/php/files/etc/php/7.2/mods-available/xdebug.ini [hunk 4]
```diff
@@ -1,7 +1 @@
-zend_extension=xdebug.so
```
Comment (stereomon) [inline saltstack/base/php/files/etc/php/7.2/mods-available/xdebug.ini:1]: This configuration is probably for older xdebug versions, if so keep the original, if not use the same as for the 8.0 one.

File: saltstack/base/php/files/etc/php/7.4/mods-available/xdebug.ini [hunk 5]
```diff
@@ -1,7 +1 @@
-zend_extension=xdebug.so
-xdebug.remote_enable=0
-xdebug.remote_host=10.10.0.1
-xdebug.remote_autostart=0
-xdebug.remote_port=9000
-# This is needed to prevent max recursion exeception when Twig templates are very complicated
-xdebug.max_nesting_level=1000
+xdebug.mode=debug
```
Comment (stereomon) [inline saltstack/base/php/files/etc/php/7.4/mods-available/xdebug.ini]: Needs to be the same as 8.0 one.

File: saltstack/base/spryker/files/etc/nginx/spryker/backendapi.conf [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        fastcgi_pass backend-$application_env-backendapi;
+        fastcgi_index index.php;
+        include /etc/nginx/fastcgi_params;
+        fastcgi_param SCRIPT_NAME /index.php;
+        fastcgi_param APPLICATION_ENV $application_env;
+        fastcgi_param APPLICATION_STORE $application_store;
+        fastcgi_param SPRYKER_PRODUCT_CONFIGURATOR_HOST demo-date-time-configurator.spryker.local;
+        fastcgi_param SPRYKER_PRODUCT_CONFIGURATOR_PORT 80;
+        fastcgi_param SPRYKER_PRODUCT_CONFIGURATOR_ENCRYPTION_KEY change123;
+        fastcgi_param SPRYKER_PRODUCT_CONFIGURATOR_HEX_INITIALIZATION_VECTOR 0c1ffefeebdab4a3d839d0e52590c9a2;
```
Comment (stereomon) [inline saltstack/base/spryker/files/etc/nginx/spryker/backendapi.conf:36]: Do we need those here?

File: saltstack/base/spryker/files/etc/nginx/spryker/backendapi.conf [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    # PHP application gets all other requests
+    location / {
+        add_header X-Server $hostname;
+        fastcgi_pass backend-$application_env-backendapi;
+        fastcgi_index index.php;
+        include /etc/nginx/fastcgi_params;
+        fastcgi_param SCRIPT_NAME /index.php;
+        fastcgi_param APPLICATION_ENV $application_env;
+        fastcgi_param APPLICATION_STORE $application_store;
+        fastcgi_param SPRYKER_PRODUCT_CONFIGURATOR_HOST demo-date-time-configurator.spryker.local;
```
Comment (stereomon) [inline saltstack/base/spryker/files/etc/nginx/spryker/backendapi.conf:33]: Shouldn't this be aware of VM_DOMAIN?

File: saltstack/base/spryker/files/etc/nginx/spryker/backendgateway.conf [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    # PHP application gets all other requests
+    location / {
+        add_header X-Server $hostname;
+        fastcgi_pass backend-$application_env-backendgateway;
+        fastcgi_index index.php;
+        include /etc/nginx/fastcgi_params;
+        fastcgi_param SCRIPT_NAME /index.php;
+        fastcgi_param APPLICATION_ENV $application_env;
+        fastcgi_param APPLICATION_STORE $application_store;
+        fastcgi_param SPRYKER_PRODUCT_CONFIGURATOR_HOST demo-date-time-configurator.spryker.local;
```
Comment (stereomon) [inline saltstack/base/spryker/files/etc/nginx/spryker/backendgateway.conf:33]: Same as above.

File: saltstack/base/spryker/files/etc/nginx/spryker/backoffice.conf [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    # PHP application gets all other requests
+    location / {
+        add_header X-Server $hostname;
+        fastcgi_pass backend-$application_env-backoffice;
+        fastcgi_index index.php;
+        include /etc/nginx/fastcgi_params;
+        fastcgi_param SCRIPT_NAME /index.php;
+        fastcgi_param APPLICATION_ENV $application_env;
+        fastcgi_param APPLICATION_STORE $application_store;
+        fastcgi_param SPRYKER_PRODUCT_CONFIGURATOR_HOST demo-date-time-configurator.spryker.local;
```
Comment (stereomon) [inline saltstack/base/spryker/files/etc/nginx/spryker/backoffice.conf:33]: Same as above

File: saltstack/base/spryker/files/etc/nginx/spryker/gateway.conf [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    # PHP application gets all other requests
+    location / {
+        add_header X-Server $hostname;
+        fastcgi_pass backend-$application_env-gateway;
+        fastcgi_index index.php;
+        include /etc/nginx/fastcgi_params;
+        fastcgi_param SCRIPT_NAME /index.php;
+        fastcgi_param APPLICATION_ENV $application_env;
+        fastcgi_param APPLICATION_STORE $application_store;
+        fastcgi_param SPRYKER_PRODUCT_CONFIGURATOR_HOST demo-date-time-configurator.spryker.local;
```
Comment (stereomon) [inline saltstack/base/spryker/files/etc/nginx/spryker/gateway.conf:33]: Same as above

File: saltstack/base/system/files/etc/motd [hunk 11]
```diff
@@ -5,5 +5,5 @@ Welcome to
   \____ \|  _ \ / ___) | | | |_/ ) ___ |/ ___)
   _____) ) |_| | |   | |_| |  _ (| ____| |
  (______/|  __/|_|    \__  |_| \_)_____)_|
-         |_|         (____/
+         |_|         (____/ dev
```
Comment (stereomon) [inline saltstack/base/system/files/etc/motd]: Why this?

**Review by stereomon — CHANGES_REQUESTED** [review]

File: saltstack/base/system/files/etc/motd (hunk 11 above)
Comment (afahmyspryker) [reply saltstack/base/system/files/etc/motd]: to know that you are sshing to dev devvm, do you want me to remove it?

File: saltstack/base/php/extensions.sls (hunk 1 above)
Comment (afahmyspryker) [reply saltstack/base/php/extensions.sls]: ok will do it

File: saltstack/base/php/files/etc/php/7.2/mods-available/xdebug.ini (hunk 4 above)
Comment (afahmyspryker) [reply saltstack/base/php/files/etc/php/7.2/mods-available/xdebug.ini:1]: ok reverting to / using: 
zend_extension=xdebug.so
xdebug.mode=off
xdebug.client_host=10.10.0.1
xdebug.client_port=9000

File: saltstack/base/php/files/etc/php/7.4/mods-available/xdebug.ini (hunk 5 above)
Comment (afahmyspryker) [reply saltstack/base/php/files/etc/php/7.4/mods-available/xdebug.ini]: done

File: saltstack/base/spryker/files/etc/nginx/spryker/backendapi.conf (hunk 6 above)
Comment (afahmyspryker) [reply saltstack/base/spryker/files/etc/nginx/spryker/backendapi.conf:36]: I really don't know, was from previous config, do you require changes or want me to remove it?

File: saltstack/base/spryker/files/etc/nginx/spryker/backendapi.conf (hunk 7 above)
Comment (afahmyspryker) [reply saltstack/base/spryker/files/etc/nginx/spryker/backendapi.conf:33]: nope, will need to create a conf for it

File: saltstack/base/php/files/etc/php/7.1/mods-available/xdebug.ini (hunk 2 above)
Comment (afahmyspryker) [reply saltstack/base/php/files/etc/php/7.1/mods-available/xdebug.ini:1]: ok

File: saltstack/base/php/files/etc/php/7.1/mods-available/xdebug.ini (hunk 3 above)
Comment (afahmyspryker) [reply saltstack/base/php/files/etc/php/7.1/mods-available/xdebug.ini:3]: ok

File: saltstack/base/system/files/etc/motd (hunk 11 above)
Comment (stereomon) [reply saltstack/base/system/files/etc/motd]: Yes, for me it looks wrong. If you want to output some informational data then it should go to the next line and should have a key for it eg. `Environment: development` but I think its not needed at all.

Comment (afahmyspryker) [reply saltstack/base/system/files/etc/motd]: ok I will remove it...

**Pull request merged**